### PR TITLE
ConnectDataWithApi

### PR DESCRIPTION
**Pull request by FedeRosaless**
image

### CODE REVIEW

File: src/screens/CryptoList/index.tsx [hunk 1]
```diff
@@ -1,48 +1,24 @@
 import {TouchableOpacity, FlatList} from 'react-native';
 import {BlueText, Container, ContainerFlatList} from './style';
-import CryptoItem from '../../components/CryptoItem';
+import CryptoItem from '../../components/cryptoItem';
```
Comment (tomasdematos) [inline src/screens/CryptoList/index.tsx]: rename cryptoItem to CryptoItem

File: src/store/reducer/RootReducer.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        `${process.env.REACT_APP_BASE_URL}${name}/metrics`,
+      ).then(res => res.json());
+      return data;
+    } catch (error) {
+      return error;
+    }
+  },
+);
+
+export const updateCrypto = createAsyncThunk(
```
Comment (tomasdematos) [inline src/store/reducer/RootReducer.ts]: This is outside the scope, It should be implemented in the last task, remove it from this PR and use it when is requested

File: src/store/reducer/RootReducer.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import {createSlice, createAsyncThunk} from '@reduxjs/toolkit';
+import {Crypto} from 'interfaces';
+
+const initialState = {
+  cryptos: [] as Crypto[],
+  loading: false,
+};
+
+export const getCrypto = createAsyncThunk(
```
Comment (tomasdematos) [inline src/store/reducer/RootReducer.ts]: the actions should be in store/actions.ts

File: src/store/reducer/RootReducer.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import {createSlice, createAsyncThunk} from '@reduxjs/toolkit';
+import {Crypto} from 'interfaces';
+
+const initialState = {
+  cryptos: [] as Crypto[],
```
Comment (tomasdematos) [inline src/store/reducer/RootReducer.ts]: you should use a type for the initialState

initialState = {
   ......
} as CryptoState

and in store/types.ts

type CryptoState {
   cryptos: Crypto[];
   loading: boolean;
   ......

}

**Comment by FedeRosaless** [comment]
image
